### ISSUE TITLE
how to issue refresh token?

### ISSUE THREAD

**Issue by sinamiandashti** [oauthjs/node-oauth2-server#3]
how the refresh token can be created?

or what is the url to issue refresh token?

or anything about it  ... plz help

**Comment by viksat7** [comment]
generaterefreshtoken :  any examples ?

**Comment by kruti-thukral** [comment]
The refresh does not seem to be working . Any inputs ?

**Comment by pedroetb** [comment]
@kruti-thukral I think that there is a bug at the (currently) stable version 3.0.1, in line 103 of [lib/grant-types/refresh-token-grant-type.js](https://github.com/oauthjs/node-oauth2-server/blob/v3.0.1/lib/grant-types/refresh-token-grant-type.js#L103). There is a hardcoded `client.id` instead of `client.clientId`. Apparently, it is fixed in [tag v4.0.0-dev.1](https://github.com/oauthjs/node-oauth2-server/blob/v4.0.0-dev.1/lib/grant-types/refresh-token-grant-type.js#L103) (but wrong again in [tag v4.0.0-dev.2](https://github.com/oauthjs/node-oauth2-server/blob/v4.0.0-dev.2/lib/grant-types/refresh-token-grant-type.js#L103)).

I'm not sure about what will be the definitive change; the property at client definition (clientId -> id) or the property at the mentioned condition (id -> clientId), so I'm not making any pull request (yet).

I have an example project using version 3.0.1 with **refresh_token** grant type working (with a fix for that bug), maybe it can help you :) [node-oauth2-server-example](https://github.com/pedroetb/node-oauth2-server-example)

**Comment by kruti-thukral** [comment]
@pedroetb - I dont see any issue with the lib code . I have directly used oauth2-server version 3.0.1
and I have the refresh token working fine.Thanks for the input though . After your comment, i realised something was going wrong with my client id and was a bug in my code . Fixing that resolved the issue.

**Comment by leapit** [comment]
> @kruti-thukral I think that there is a bug at the (currently) stable version 3.0.1, in line 103 of [lib/grant-types/refresh-token-grant-type.js](https://github.com/oauthjs/node-oauth2-server/blob/v3.0.1/lib/grant-types/refresh-token-grant-type.js#L103). There is a hardcoded `client.id` instead of `client.clientId`. Apparently, it is fixed in [tag v4.0.0-dev.1](https://github.com/oauthjs/node-oauth2-server/blob/v4.0.0-dev.1/lib/grant-types/refresh-token-grant-type.js#L103) (but wrong again in [tag v4.0.0-dev.2](https://github.com/oauthjs/node-oauth2-server/blob/v4.0.0-dev.2/lib/grant-types/refresh-token-grant-type.js#L103)).
> 
> I'm not sure about what will be the definitive change; the property at client definition (clientId -> id) or the property at the mentioned condition (id -> clientId), so I'm not making any pull request (yet).
> 
> I have an example project using version 3.0.1 with **refresh_token** grant type working (with a fix for that bug), maybe it can help you :) [node-oauth2-server-example](https://github.com/pedroetb/node-oauth2-server-example)

@pedroetb I tried your example,still error when use refresh_token type,got below error
```
curl http://localhost:3000/oauth/token \
> -d "grant_type=refresh_token" \
> -d "client_id=application" \
> -d "client_secret=secret" \
> -d "refresh_token=5de9099aa1925063d05b75c4cca4377c9f9982ea" \
> -H "Authorization: Basic YXBwbGljYXRpb246c2VjcmV0" \
> -H "Content-Type: application/x-www-form-urlencoded"
```
```
{"statusCode":400,"status":400,"code":400,"message":"Unauthorized client: `grant_type` is invalid","name":"unauthorized_client"}
```

**Comment by pedroetb** [comment]
@leapit discussed and fixed at https://github.com/pedroetb/node-oauth2-server-mongo-example/issues/7
